integrationtesting/opendcs-tests/src/test/java/org/opendcs/regression_tests/DBImportExportTest.java

```diff
@@ -0,0 +1,124 @@
+package org.opendcs.regression_tests;
+
+import decodes.dbimport.DbExport;
+import decodes.dbimport.DbImport;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.opendcs.fixtures.AppTestBase;
+import org.opendcs.fixtures.annotations.DecodesConfigurationRequired;
+import org.opendcs.fixtures.helpers.TestResources;
+import org.opendcs.spi.configuration.Configuration;
+import org.opendcs.utils.logging.OpenDcsLoggerFactory;
+import org.slf4j.Logger;
+import uk.org.webcompere.systemstubs.SystemStubs;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@DecodesConfigurationRequired({
+        "DBImportExportTest/existing.xml"
+})
+public class DBImportExportTest extends AppTestBase
+{
+    private static final Logger log = OpenDcsLoggerFactory.getLogger();
+
+    @ParameterizedTest
+    @CsvSource(useHeadersInDisplayName = true, value = { // would check datasources here too, but they're always zero in the DB for some reason
+        "Name, Expected DBExport File, Expected Platforms, Expected Routing Specs",
+        "normal,DBImportExportTest/goldennormal${impl}.xml,3,2",
+        "platonly,DBImportExportTest/goldenplatonly${impl}.xml,3,1",
+        "platover,DBImportExportTest/goldenplatover${impl}.xml,2,1",
+    })
+    public void test_normal(String type, String expectedResultFile, int plats, int rs) throws Exception
+    {
+        Configuration config = this.configuration;
+        File logFile = new File(config.getUserDir(),"/dbimport-" + type + ".log");
+
+        assertExitNullOrZero();
+
+        String clearDB = SystemStubs.tapSystemOut(
+                () -> exit.execute(() ->
+                        DbImport.main(
+                                args("-l", logFile.getAbsolutePath(), "-W", "-y",
+                                        TestResources.getResource(configuration,"DBImportExportTest/existing.xml"))
+                        )
+                )
+        );
+        assertExitNullOrZero();
+
+        List<String> platargs =  new ArrayList<>(Arrays.asList("-l", logFile.getAbsolutePath(), "-d3"));
+        switch(type) {
+            case "normal":
+                log.info("Loading platform, routing spec, etc with no extra flags.");
+                break;
+            case "platonly":
+                log.info("Loading platforms with platform only flag.");
+                platargs.add("-p");
+                break;
+            case "platover":
+                log.info("Loading platforms with platform only and overwrite flags.");
+                platargs.addAll(Arrays.asList("-W", "-y", "-p"));
```

Comment:
Using `-W` here and above is the most likely issue, I don't believe the system is setup to do anything *other* than a total wipe.